src/main.zig

```diff
@@ -116,40 +116,45 @@ fn printResults(allocator: std.mem.Allocator, results: *std.ArrayList(SearchResu
 
     var hasMatches = false;
 
-    // reserve some conservative amount of memory to avoid initial allocs
-    var buffer = try std.ArrayList(u8).initCapacity(allocator, 8 * 1024);
-    defer buffer.deinit();
+    // get a buffered writer for stdout
+    const stdout = std.io.getStdOut();
+    var bw = std.io.BufferedWriter(8 * 1024, std.fs.File.Writer){ .unbuffered_writer = stdout.writer() };
+    defer stdout.close();
+
+    const colorConfig = std.io.tty.detectConfig(stdout);
```

Comment:
Could be created only once needed right (I think it makes a syscall on windows)?